### ISSUE TITLE
Document Waku's Value Proposition and USP

### ISSUE THREAD

**Issue by fryorcraken** [waku-org/docs.waku.org#134]
Seeing the interested generated by Waku at the recent events, it seems that we need to more strongly convey the USP and value proposition of Waku.
Something to be more generalized and clearer than https://docs.waku.org/overview/use-cases

Attempting to provide a first draft below.

---

## The Waku Network

The Waku Network is a set of nodes that enable exchange small payloads across users of an application. It is
- permissionless: anyone can run a node, join the network and be discovered 
- DOS projected: Rate limitation applies to sending message, ensuring reasonable bandwidth usage and preventing users with lower bandwidth to be kicked out of the network
- scalable: the network is split in shards, enabling network traffic to increase beyond the capacity of individual nodes.
- a service network: nodes can run services to serve users with limited bandwidth or devices that are mostly offline

The parameters of the current Waku Network (Gen 0) are defined in this RFC: https://rfc.vac.dev/spec/64/

Let's review how The Waku Network can help your application:

## Peer-To-Peer-Network-As-A-Service

The Waku Network can be used as an existing peer-to-peer network, enabling your project to leverage properties of such network without building your own.

### To Build No-Infra DApps

You can leverage the Waku Network the same way you would leverage the Ethereum blockchain: to be able to build no infra dApp. By deploying a smart contract on Ethereum and publish a webapp to use said contract, you can build a dApp with no or minimal infra (just hosting your webapp, using centralized or decentralized tools).

With Waku, you can do the same and enable interaction between users without having to run your own database or backend.

Do note that similarly to Ethereum, where users pay gas fees to use a smart contract, users will also have to pay for the usage of Waku. Currently, this means acquiring a RLN membership to send message or paying a store service to retrieve missed messages (for being offline)

Status: WIP
- The Waku Network is only being rolled out now, with smart contracts on Ethereum testnet Sepolia
- Paying for store service (retrieve messages) is still under research

### To Build Web only Dapps

Thanks to the no-infra value proposition, it is also possible to build a decentralized webapp that leverage Waku as a peer-to-peer network, without deploying infra, apart maybe for hosting the webapp (webserver only).

### To Bootstrap a Peer-to-Peer Network

If you intend to build a peer-to-peer network, for which decentralization is important. This could be for privacy or censorship-resistant reason, then you may not want to bootstrap the network by running all the nodes yourself, until your community grows.

In this instance, the Waku Network can be used as the peer-to-peer communication layer of your project.
Nodes can use the wider Waku network to exchange data.
Once your own application is more broadly used, and is capable of the level of decentralization you need, you may get off the Waku Network and use your own (Waku) p2p network.

Status: WIP
- The Waku Network has only started and the number of Waku node may not yet be sufficient for this purpose.
- Once incentivization is in place, we can expect the network to grow more as node operators will be running for monetary reasons.

### As a Signal Network

The Waku Network is optimized to provide fair/low latency data exchange across a peer-to-peer network.
This may not be suitable for higher bandwidth usage such as streaming, video call or large data transfers.

In this instance, the Waku Network can still be used to enable these use cases in a no-infra/decentralized/privacy friendly manner. Handshake (SDP, noise, etc) can be done over Waku to enable direct WebRTC, TCP, holepunching connections between 2 or more users.

TBC

**Comment by fryorcraken** [comment]
Need to mature, I think there is something to say about native Web3 technology and integrating with Web3 tech.

Thinking about a case where customer does not care about several properties: decentralization, censorship resistance and privacy.
However, thanks to Waku SDK, it may be easier to integrate Waku than another Web2 tasks.

[70/ETH-SECPM](https://rfc.vac.dev/spec/70/) is a good example. Assuming we have the right docs, and SDKs, then this gives an easy way for users to use their Ethereum or other blockchain account as identity and entry point in the Waku network/protocol.

**Comment by s-tikhomirov** [comment]
> users using relay node benefit of k-anonymity in terms of application usage.

Isn't any anonymity k-anonymity? For a skeptical reader, k-anonymity may sound like a deliberately complex-sounding term, when in fact what we're saying is rather standard: the more users a network has, the larger its _theoretical_ anonymity set. Also, whether strong anonymity is achieved in practice depends on many implementation details; even if `k` is large, it doesn't automatically follow that

> external parties or even other nodes in the network, will not be able to determine the applications used by the user.

**Comment by vpavlin** [comment]
> Thanks to using a shared network for all applications, users using relay node benefit of k-anonymity

This is something I always struggle with - we add content-topic to every message and we even tell app creators to name them appropriately (i.e. with some app identification - e.g. name), so is using k-anonymity actually per shard or per content-topic? 

> To Bootstrap a Peer-to-Peer Network

This is great - I even got a question of "why should I use Waku if I can simply create my own network?"...well..can you? When? For how much?:) So positioning Waku as a reasonable bootstrap for you app sounds like a great way to get into people minds.

> Waku for node operators

does this belong in the list of USPs?

> Using zeroknowledge technology, users do not dox their blockchain address when sending a rate-limited message.

We need to find a better wording because it basically says "user needs to do a TX" (which means doxing) and then "using ZK to not dox).

I understand the not doxxing is related to sending the messages, but the concern of doxxing the address during registration is already there 

> Thanks to anonymated rate limit

anonymous? anonymized?

> This can be the case when there is no possible consensus

This **cannot** be the case... ?

> * Asymmetric

symmetric?


> [70/ETH-SECPM](https://rfc.vac.dev/spec/70/) is a good example.

IMO this plays into "interoperability" reasoning - you can use all your "achievements" (POAPs, NFTs, memberships..) from EVM blockchains (because we use the same key / address schemas)  and leverage it in off-chain world? Maybe?

**Comment by fryorcraken** [comment]
> Isn't any anonymity k-anonymity? For a skeptical reader, k-anonymity may sound like a deliberately complex-sounding term, when in fact what we're saying is rather standard: the more users a network has, the larger its theoretical anonymity set. Whether strong anonymity is achieved in practice depends on many implementation details; even if k is large, it doesn't automatically follow that

I agree and I think we need to be careful with the terminology we use in our communication. There is indeed a fine balance between getting the message through (anonymity) and providing accurate information (k-anonymity).

**Comment by fryorcraken** [comment]
> This is something I always struggle with - we add content-topic to every message and we even tell app creators to name them appropriately (i.e. with some app identification - e.g. name), so is using k-anonymity actually per shard or per content-topic?

Both :)

When using relay, and as long as there are enough users in the shards and enough different type of dApps, it is not possible to know:
- what node send a message with content topic `mydapp`
- what node processes messages with content topic `mydapp`

However, this is lost as soon as store, filter or light push are used: the service node knows that this specific node/ip is using `mydapp`, or that this node is the originator of the message (with light push, except if tor push is used for light push https://rfc.vac.dev/spec/47/).

There is a second level of k-anonymity in the usage of content topics, to still get _some_ anonymity_ when using req-res protocols. That is in the usage of the content topic.

If content topic are formed as follows: `/mydapp/0/<eth address of recipient>/proto`.

Then,
- when using light push, service node learns this peer (ip) is trying to contact owner of "eth address"
- when using filter or store, service node learns this peer (ip) is the owner of "eth address" (time to grab the wrench).

Instead, what [Status](https://specs.status.im/spec/10#contact-code-topic) does and what we should recommend our user, is to make content topics a bucket of k-size to still have some anonymity for users of req-res protocols.

With the current design of auto-sharding, you cannot hid what app you are using. But you can hide some metadata.

Calculate code topic: `hash(<eth address of recipent)` and take the first 4 bytes.

Change the content topic to 

`/mydapp/0/<code topic>/proto`.

Now, the service node does not learn the ethereum address of the user. It only learns that the eth address is one of those whose hash is the first bytes set in the code topic.

**Comment by fryorcraken** [comment]
> > Waku for node operators
> 
> does this belong in the list of USPs?

Well yes, operators are also part of our audience. This post is not specific to project/developers only.

**Comment by weboko** [comment]
@fryorcraken do we need to follow up on this issue with findings from Athen's offsite?

**Comment by fryorcraken** [comment]
> @fryorcraken do we need to follow up on this issue with findings from Athen's offsite?

Yes, I intend to write a blog post to discuss about Waku's offering and portfolio. or Maybe a Vac forum post.
